### ISSUE TITLE
P2P: don't relay a tx if we already have it

### ISSUE THREAD

**Issue by jcnelson** [blockstack/stacks-blockchain#1943]
The relayer will relay a tx even if it's already stored.  This is because of this offending line in `store_transaction()`, reproduced below:

```
    /// Store a single transaction
    /// Return true if stored; false if it was a dup
    fn store_transaction(
        mempool: &mut MemPoolDB,
        consensus_hash: &ConsensusHash,
        block_hash: &BlockHeaderHash,
        tx: StacksTransaction,
    ) -> bool {
        let txid = tx.txid();
        if mempool.has_tx(&txid) {
            return true;  // OOPS! This should return `false`
        }

        if let Err(e) = mempool.submit(consensus_hash, block_hash, tx) {
            info!("Reject transaction {}: {:?}", txid, &e);
            return false;
        }

        debug!("Stored tx {}", txid);
        return true;
    }
```

**Comment by jcnelson** [comment]
This is related to #1936

**Comment by jcnelson** [comment]
Fixed in #1903
